### PR TITLE
Show scroll bar on hover only

### PR DESCRIPTION
**Pull request by tudi2d**
@benjdlambert 

Follow up PR to #6344 as mentioned here https://github.com/backstage/backstage/pull/6344#issuecomment-874576937 to support similar behaviour across browsers. 

It looks like this in **Chrome** now: 

sidebar_hover

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/core-components/src/layout/Sidebar/Items.tsx [hunk 1]
```diff
@@ -299,12 +299,17 @@ export const SidebarScrollWrapper = styled('div')(({ theme }) => ({
   // Display at least one item in the container
   // Question: Can this be a config/theme variable - if so, which? :/
   minHeight: '48px',
-  '&::-webkit-scrollbar': {
+  overflowY: 'hidden',
+
+  '&:hover': {
```
Comment (freben) [inline packages/core-components/src/layout/Sidebar/Items.tsx:304]: only question here i guess is how this affects touch devices; would one want it to always be visible on those 🤔 

this stuff is hard :P in any case this is a nice improvement and we could always address that separately

**Review by freben — APPROVED** [review]

File: packages/core-components/src/layout/Sidebar/Items.tsx (hunk 1 above)
Comment (benjdlambert) [reply packages/core-components/src/layout/Sidebar/Items.tsx:304]: That's a good shout, I seem to remember that they might always be visible on touch devices by default, but I could be wrong and we should check this to confirm in a a trusty iPhone sim :)

Comment (tudi2d) [reply packages/core-components/src/layout/Sidebar/Items.tsx]: ahhhh - good catch. This always happens 🙆‍♂️ Should work now.

**Review by freben — APPROVED** [review]
NICE

**Comment by freben** [comment]
man you're killing it in these parts, so happy to have some professional touch (ha!) being added to up the polish levels

**Comment by tudi2d** [comment]
Thanks @freben - appreciate it! Really enjoy this kind of things! :)

**Pull request merged**